crates/sqlexec/src/planner/session_planner.rs

```diff
@@ -497,6 +498,28 @@ impl<'a> SessionPlanner<'a> {
                     compression: compression.map(|c| c.to_string()),
                 })
             }
+            TableOptions::DELTA | TableOptions::ICEBERG => {
+                let location = m.remove_required("location")?;
+
+                let mut storage_options = StorageOptions::try_from(m)?;
+                if let Some(creds) = creds_options {
+                    storage_options_with_credentials(&mut storage_options, creds);
+                }
+
+                // TODO: Perform a HEAD request to validate the provided config options?
```

Comment:
I sort of feel like we should do this lazily (e.g. not check until it gets used to avoid having an error on startup that would cause it to get stuck), but maybe provide an option to validate the catalog which would just connect to everything as another operation/report.

Reply:
Yeah, makes sense. In any case, I'll leave it for another PR as this one is getting quite large.

Reply:
sounds good.... does it make sesne file an issue for this kind of thing? I always feel like code TODOs make bad project management tools 😂

Reply:
Fair point, opened https://github.com/GlareDB/glaredb/issues/1817 to track that.

Reply:
Follow-up PR for that issue: https://github.com/GlareDB/glaredb/pull/1819